share/spice/twitter/twitter.js

```diff
@@ -1,132 +1,59 @@
-	var display_url = DDG.getProperty(item, 'entities.url.urls.0.display_url');
-
-	return [display_url, expanded_url];
-    }
-
-    // Display the plugin.
-    Spice.add({
-        id: 'twitter',
-        data                     : api_result,
-        name: 'Social',
-	signal: 'high',
-	meta: {
-            sourceUrl               : "https://twitter.com/" + api_result.user,
-            sourceName              : "Twitter",
-        },
-        normalize: function(item) {
-	    var urls = getURL(item);
-
-            return {
-                image: bigger_picture(item.profile_image),
-		display_url: urls[0],
-		expanded_url: urls[1]
-            };
-        },
-        templates: {
-	    group: 'base',
-	    options: {
-		content: Spice.twitter.content,
-		moreAt: true
-	    }
+        if(!api_result || (!api_result.current_status && !api_result.description)) {
+            return Spice.failed('twitter');
         }
-        
-    });
-};
 
-Handlebars.registerHelper("twitter_findLinks", function(text, entities, options) {
-    "use strict";
-
-    // Chop the string so that we can surreptitiously insert links.
-    var twitterSplit = function(twitter, result, final_text, original, start_index, i) {
-        if(twitter.length === i || twitter.length === 0) {
-            result.push({text: final_text});
-            return result;
-        } else {
-            var indices = twitter[i].indices;
-            // Text only.
-            result.push({
-                text: original.substring(start_index, indices[0])
-            });
-            // Text with link.
-            result.push({
-                text: original.substring(indices[0], indices[1]), 
-                link: twitter[i]
-            });
-            start_index = indices[1];
-            return twitterSplit(twitter, result, original.substring(start_index, original.length), original, start_index, i + 1);
+        // Fetch bigger profile image
+        var bigger_picture = function(image) {
+            return image.replace(/_normal\./, "_bigger.");
         }
-    };
 
-    // Concatenate all the entities. (It's easier this way.)
-    var all_entities = [];
-    for(var k in entities) {
-        if(entities.hasOwnProperty(k)) {
-            all_entities = all_entities.concat(entities[k]);
-        }
-    }
-    
-    // Sort indices. (These things have to be in order.)
-    all_entities.sort(function(a, b) {
-        if(a.indices[0] < b.indices[0]) {
-            return -1;
-        }
-        if(a.indices[0] > b.indices[0]) {
-            return 1;
+        // Get array of URL text and URL
+        function getURL(item) {
+            var url = DDG.getProperty(item, 'entities.url.urls.0.url');
+            var text = DDG.getProperty(item, 'entities.url.urls.0.display_url');
+            if (url) {
+                return [{
+                   text: text,
+                   href: url
+                }];
+            } else {
+                return "No website provided";
```

Comment:
Meaning, don't say that there's nothing, the lack of it being there should signal that ;)